tests/ImageKit/Upload/UploadTest.php

```diff
@@ -60,144 +64,336 @@ private function stubHttpClient($response)
+            "useUniqueFileName" => false,                                        // true|false
+            "tags" => implode(',',["abd", "def"]),                              // Comma Separated, Max length: 500 chars
+            "folder" => "/sample-folder",                                          // Using multiple forward slash (/) creates a nested folder
+            "isPrivateFile" => true,                                           // true|false
+            "customCoordinates" => implode(",", ["10", "10", "100", "100"]),    // Comma Separated, Max length: 500 chars
+            "responseFields" => implode(",", ["tags", "customMetadata"]),       // Comma Separated, check docs for more responseFields
+            "extensions" => [                                                  // An array of extensions, for more extensions refer to docs
+                [
+                    "name" => "remove-bg",
+                    "options" => [  // all parameters inside this object are sent directly to the third-party service
+                        "add_shadow" => true
+                    ]
+                ]
+            ],
+            "webhookUrl" => "https://example.com/webhook",                      // Notification URL to receive the final status of pending extensions
+            "overwriteFile" => true,                                            // true|false, in case of false useUniqueFileName should be true
+            "overwriteAITags" => false,                                          // true|false, set to false in order to preserve overwriteAITags
+            "overwriteTags" => false,                                            // true|false
+            "overwriteCustomMetadata" => true,                                  // true|false
+            "customMetadata" => [                                              // An array of created custom fields, for more details refer to docs
+                    "SKU" => "VS882HJ2JD",
+                    "price" => 599.99,
+            ]
+        ];
+
+        $mockBodyResponse = Utils::streamFor(json_encode($this->uploadSuccessResponseObj));
+
+        $mock = new MockHandler([
+            new Response(200, ['X-Foo' => 'Bar'], $mockBodyResponse)
+        ]);
 
-        $this->stubHttpClient(new Response(200, ['X-Foo' => 'Bar'], $mockBodyResponse));
+        $handlerStack = HandlerStack::create($mock);
 
-        $res = $this->client->upload([
-            'file' => 'http://lorempixel.com/640/480/',
-            'fileName' => '7e57d004-2b97-0e7a-b45f-5387367791cd',
+        $container = [];
+        $history = Middleware::history($container);
+
+        $handlerStack->push($history);
+        
+        $client = new Client(['handler' => $handlerStack]);
+
+        $requestMultiPart = GuzzleHttpWrapper::getMultipartData($fileOptions);
+        $client->request('POST', Endpoints::getUploadFileEndpoint(), [
+            'multipart' =>$requestMultiPart
         ]);
```

Comment:
Why are we doing this here instead of calling `$this->client->uploadFile($fileOptions);` We need to assert that request generated by calling `$this->client->uploadFile($fileOptions);` is correct. Here we are generating request in the test case differently and it has nothign to do with implementation of `$this->client->uploadFile($fileOptions);`